### PR TITLE
Rename EnableOverridesUx to avoid confusion

### PR DESCRIPTION
**Pull request by forkercat**
## What does this PR do?

This PR renames the prefab override feature flag `EnableOverridesUx` to `EnableOutlinerOverrideManagement`. We recently introduced a new flag `EnableInspectorOverrideManagement` for override feature in DPE Inspector. Renaming here can help reduce confusion.

```
"O3DE": {
  "Preferences": {
    "Prefabs": {
      "EnableOverridesUx": false  // changed to "EnableOutlinerOverrideManagement"
    }
  }
}
```

Outliner Override RFC: https://github.com/o3de/sig-content/issues/75

## How was this PR tested?

- [x] Passed prefab automated tests
- [x] Updated default settings in AutomatedTesting/Registry
- [x] Send out impactful notice in Discord channel
- [ ] Update couple of docs accordingly
- [x] Update [public-facing docs](https://development--o3deorg.netlify.app/docs/learning-guide/tutorials/entities-and-prefabs/override-a-prefab/) for override accordingly: https://github.com/o3de/o3de.org/pull/2294

### CODE REVIEW

**Comment by forkercat** [comment]
TODO: Need to do a rebase after some prefab PRs are merged.

**Review by srikappa-amzn — COMMENTED** [review]
There needs to be an impactful change discord notice about this. 
Also, if the value is hardcoded (probably to false) in o3de or any project repos, the name needs to be updated in those places too

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabEditorPreferences.h [hunk 1]
```diff
@@ -12,12 +12,18 @@
 
 namespace AzToolsFramework::Prefab
 {
-    extern const AZStd::string_view EnablePrefabOverridesUxKey;
-    extern const AZStd::string_view InspectorOverrideManagementKey;
     extern const AZStd::string_view HotReloadToggleKey;
-
+    extern const AZStd::string_view OutlinerOverrideManagementKey;
+    extern const AZStd::string_view InspectorOverrideManagementKey;
```
Comment (srikappa-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabEditorPreferences.h:17]: nit : name order.

**Comment by forkercat** [comment]
> Also, if the value is hardcoded (probably to false) in o3de or any project repos, the name needs to be updated in those places too
I will do a double check.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabEditorPreferences.cpp [hunk 2]
```diff
@@ -11,9 +11,9 @@
 
 namespace AzToolsFramework::Prefab
 {
-    const AZStd::string_view EnablePrefabOverridesUxKey = "/O3DE/Preferences/Prefabs/EnableOverridesUx";
-    const AZStd::string_view InspectorOverrideManagementKey = "/O3DE/Preferences/Prefabs/EnableInspectorOverrideManagement";
     const AZStd::string_view HotReloadToggleKey = "/O3DE/Preferences/Prefabs/EnableHotReloading";
+    const AZStd::string_view OutlinerOverrideManagement = "/O3DE/Preferences/Prefabs/EnableOutlinerOverrideManagement";
```
Comment (michabr) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabEditorPreferences.cpp:15]: Should this end in Key like the others?

**Review by michabr — COMMENTED** [review]
Nice. So can EnableInspectorOverrideManagement be enabled and working without the EnableOutlinerOverrideManagement flag enabled?

**Comment by forkercat** [comment]
> So can EnableInspectorOverrideManagement be enabled and working without the EnableOutlinerOverrideManagement flag enabled?

Yes. `EnableInspectorOverrideManagement` can be enabled and working without the other flag enabled. But the users would not be able to select an overridden entity since container is not expanded.

**Comment by forkercat** [comment]
> There needs to be an impactful change discord notice about this.
> Also, if the value is hardcoded (probably to false) in o3de or any project repos, the name needs to be updated in those places too

I will send out an impactful change notice in Discord. I've verified that in o3de or any project repos there is no `EnableOverridesUx` key present and hardcoded.

**Comment by forkercat** [comment]
@michabr Let me know if you still have any question. Otherwise, I will merge the PR on 3/2.

**Review by michabr — APPROVED** [review]

**Pull request merged**